sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1058,37 +1059,66 @@ private static void validateMethodAnnotations(
 
     // Verify that no setter has @JsonIgnore.
-    validateSettersNoAnnotations(
-        methodNameToAllMethodMap, descriptors, AnnotationPredicate.JSON_IGNORE);
+    validateSettersDoNotHaveAnnotation(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicates.JSON_IGNORE);
 
     // Verify that no setter has @Default.
-    validateSettersNoAnnotations(
-        methodNameToAllMethodMap, descriptors, AnnotationPredicate.DEFAULT_VALUE);
+    validateSettersDoNotHaveAnnotation(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicates.DEFAULT_VALUE);
   }
 
   /**
    * Validates that getters don't have mixed annotation.
    */
-  private static void validateGettersNoMixedAnnotations(
+  private static void validateGettersHaveConsistentAnnotation(
       SortedSetMultimap<Method, Method> methodNameToAllMethodMap,
       List<PropertyDescriptor> descriptors,
-      AnnotationPredicate annotationPredicate) {
-    List<InconsistentlyAnnotatedGetters> incompletelyAnnotatedGetters = new ArrayList<>();
-    for (PropertyDescriptor descriptor : descriptors) {
+      final AnnotationPredicates annotationPredicates) {
+    List<InconsistentlyAnnotatedGetters> inconsistentlyAnnotatedGetters = new ArrayList<>();
+    for (final PropertyDescriptor descriptor : descriptors) {
       if (descriptor.getReadMethod() == null
           || IGNORED_METHODS.contains(descriptor.getReadMethod())) {
         continue;
       }
+
       SortedSet<Method> getters = methodNameToAllMethodMap.get(descriptor.getReadMethod());
-      SortedSet<Method> gettersWithTheAnnotation = Sets.filter(getters, annotationPredicate);
+      SortedSet<Method> gettersWithTheAnnotation =
+          Sets.filter(getters, annotationPredicates.forMethod);
+      Set<Annotation> propertyAnnotations = Sets.newLinkedHashSet(FluentIterable
+          .from(gettersWithTheAnnotation)
+          .transform(new Function<Method, Annotation>() {
+            @Nullable
+            @Override
+            public Annotation apply(@Nonnull Method method) {
+              FluentIterable<Annotation> methodAnnotations = FluentIterable
+                  .of(method.getAnnotations())
+                  .filter(annotationPredicates.forAnnotation);
+              int annotationCount = methodAnnotations.size();
+              if (annotationCount == 0) {
+                return null;
+              } else if (annotationCount == 1) {
+                return methodAnnotations.get(0);
+              } else {
+                throw new IllegalArgumentException(String.format(
+                    "Method [%s] is marked with multiple annotations: %s.",
+                    method.getName(),
+                    methodAnnotations));
+              }
+            }})
+          .filter(Predicates.notNull()));
+
+      checkArgument(propertyAnnotations.size() <= 1,
+          "Property [%s] is marked with multiple annotations: %s.",
```

Comment:
the error isn't that there are multiple annotations but they aren't all equivalent.
with multiple annotations -> with contradictory annotations

also, your going to make people search for where they are if you don't include the mapping from annotation to methods.

Something like:
Property [%s] is marked with contradictory annotations. Found [annotation1] on [methods], [annotation2] on [other methods], ...